### DISCUSSION TITLE
Edit DataGrid pointerover style

### DISCUSSION THREAD

**Discussion by vadd98** [AvaloniaUI/Avalonia#5692] [Q&A]
I'm implementing a DataGrid in my app. I'd like to change the background and foreground color of my rows when the specific row enters the pointerover state.
At the moment I already have successfully changed the pointerover background color doing this:

```
                <Style Selector="DataGridRow:pointerover /template/ Rectangle">
                    <Setter Property="Fill" Value="Blue"/>
                </Style> 
```

By the way, I can't find any way to change the text foreground color.
I'm actually looking at this xaml [https://github.com/AvaloniaUI/DataGrid/blob/master/src/Avalonia.DataGrid.Themes.Default/DefaultTheme.xaml](url) and this xaml [https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Controls.DataGrid/Themes/Fluent.xaml](url) for reference.

Any suggestion?

**Comment by maxkatz6** [comment]
Try ```Property"TextBlock.Foreground"``` on either DataGridRow or DataGridCell controls.

**Reply by vadd98** [reply]
Thans, that worked!
Another question: what is the control that manages the editing mode of a Cell? I'd like to customise its background, stroke and foreground too

**Reply by maxkatz6** [reply]
It's simple TextBox.
https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Controls.DataGrid/DataGridTextColumn.cs#L144

**Reply by vadd98** [reply]
Thanks man, that was it.
Last question: is it normal that the ScrollBar in the DataGrid appears only if I set `VerticalScrollBarVisibility="Visible"` while if I set `VerticalScrollBarVisibility="Auto"` it never shows? I underline that scrolling works with Auto too using the mouse wheel, even though the ScrollBar is not visible.

**Reply by maxkatz6** [reply]
DataGrid works in weird way with scrolling, so I assume it's not expected, but not surprising either.

**Reply by nightking4911** [reply]
Does anyone know how do remove the default styling of data grid like when we select a row its text content turns to white ?
